src/utils/setupHelper.js

```diff
@@ -0,0 +1,146 @@
+  //get the data for the players in the game
+  let players = await gameState.collection('players').get().then(getSnapshotData);
+
+  let playersDocs = await getCollectionDocs(gameState, 'players');
+  //update each player's location to Atlanta
+  //shuffle the roles & assign them
+  const gameRoles = shuffle(roles);
+  await playersDocs.forEach((player, i) => {
+    player.ref.update('currentCity', 'Atlanta');
+    player.ref.update('role', gameRoles[i]);
+  });
+
+  return {players, playersDocs};
+}
+
+//distribute cards to each player depending on the number of players
+export async function distributeCards(gameState, playerDeck, players, playersDocs) {
+  const numPlayers = Object.keys(players).length;
+  const numCards = cardsPerPlayer(numPlayers);
+  for (let i = 0; i < numPlayers; i++) {
+    let currentHand = [];
+    for (let j = 0; j < numCards; j++) {
+      currentHand.push(playerDeck.pop());
+    }
+    await playersDocs[i].ref.set({currentHand}, {merge: true})
+  }
+}
+
+//calculate how many cards per player we're getting
+function cardsPerPlayer(numPlayers) {
+  if (numPlayers === 2 ) return 4;
+  if (numPlayers === 3) return 3;
+  if (numPlayers === 4) return 2;
+}
+
+/*-------------------CREATE PLAYER DECK ----------------------*/
+//decide on number of cards -> initial difficulty Level was a string of "Beginner", "Intermediate", and "Heroic".
+export function getNumCards(difficultyLevel) {
+  if (difficultyLevel === 4) {
+    return 4;
+  }
+  else if (difficultyLevel === 5) {
+    return 5;
+  }
+  else if (difficultyLevel === 6) {
+    return 6;
+  }
+}
+
+export async function createPlayerDeck(gameState, players, difficultyLevel, playersDocs) {
+  const numEpidemicCards = getNumCards(difficultyLevel);
+  let playerDeck = [];
+
+  //get the docs to get their references
+  const cities = await getCollectionDocs(gameState, 'unusedCityCards');
+  const events = await getCollectionDocs(gameState, 'unusedEventCards');
+
+  //add the city and event cards into the deck, and shuffle the deck.
+  cities.forEach(city => playerDeck.push(city.ref));
+  events.forEach(event => playerDeck.push(event.ref));
+  playerDeck = shuffle(playerDeck);
+
+  //distribute x number of cards to each player
+  await distributeCards(gameState, playerDeck, players, playersDocs);
+
+  const epidemics = await getCollectionDocs(gameState, 'epidemicCards');
+  const epCards = epidemics.slice(0, numEpidemicCards).map(epidemic => epidemic.ref);
+
+  playerDeck = splitShuffle(playerDeck, numEpidemicCards, epCards);
+  await gameState.set({playerDeck}, {merge: true});
+}
+
+//separate the cards in numEpidemicCards pile.
+//shuffle those cards with the epidemic card.
+//put them back together as your new deck.
+function splitShuffle(playerDeck, numPiles, epCards) {
+  const separatePiles = [];
+  playerDeck.forEach((card, i) => {
+    separatePiles[i % numPiles] = separatePiles[i % numPiles] || [];
```

Comment:
Can do `separatePiles[i % numPiles] = separatePiles[i % numPiles] ? separatePiles[i % numPiles].push(card) : []` but actually I'm not sure which is clearer...